airbrake/notice.py

```diff
@@ -44,38 +44,35 @@ def __init__(self, exception, params=None, session=None,
             data = _trim_data(user, ['id', 'name', 'email'])
             if not context:
                 self.context = {}
-            self.context["user"] = data
+            self.context['user'] = data
 
         self.notifier = notifier
 
-        if isinstance(exception, str):
-            error = {
-                'type': "Error",
-                'backtrace': [{'file': "N/A",
-                               'line': 1,
-                               'function': "N/A"}],
-                'message': exception,
-                'severity': severity or ErrorLevels.DEFAULT_LEVEL}
-            self.errors = [error]
+        exc_info = sys.exc_info()
 
-        if isinstance(exception, (BaseException, Exception)):
+        # First try to use an already-created Error
+        if isinstance(exception, Error):
+            self.errors = [exception.data]
+
+        # Next try to create an Error out of the stacktrace (if available).
+        elif exc_info and exc_info[0]:
+            err = Error(exc_info)
+            self.errors = [err.data]
```

Comment:
Yup, great points and I agree with you guys. I've created https://github.com/airbrake/airbrake-python/issues/79 to deal with this in more detail